src/common/utils.js

```diff
@@ -352,3 +352,22 @@ module.exports.getEthereumBlockNumber = web3 => {
     web3.eth.getBlockNumber((error, data) => (error ? reject(error) : resolve(data)));
   });
 };
+
+/**
+ * Transform a list of states to a corresponding map.
+ * @param {Array<Object>} statesList - list of states in the format {address, key, data}
+ * @return {Object} a map whose keys are addresses and each object is a map of states (key=>data)
+ * */
+module.exports.transformStatesListToMap = statesList => {
+  const statesMap = {};
+  for (let i = 0; i < statesList.length; ++i) {
+    const address = JSON.stringify(statesList[i].address);
+    if (!(address in statesMap)) {
+      statesMap[address] = {};
+    }
+    const key = statesList[i].key;
+    const delta = statesList[i].data;
+    statesMap[address][key] = delta;
+  }
```

Comment:
Must verify that `JSON.stringify` is consistent.
E.g. that `JSON.stringify({a:1,b:2})` always outputs `{"a":1,"b":2}` and never `{"b":2,"a":1}` (or the other way around).

![Screenshot from 2019-12-25 09-52-44](https://user-images.githubusercontent.com/22766533/71437967-66691c00-26fc-11ea-8474-cd8742347583.png)

Also: https://stackoverflow.com/questions/42491226/is-json-stringify-deterministic-in-v8

Reply:
A possible solution: https://www.npmjs.com/package/json-stringify-deterministic